doc/developer_guide/developer_guide.md

```diff
@@ -1,7 +1,116 @@
 # Developer Guide
 
 
-In this developer guide we explain how you can build this project.
+In this developer guide we explain how you can build and run this project.
 
-* [building_documentation](building_documentation.md)
 
+
+## Building the Project
+
+### 1. Build the Python Package
+This project needs python interpreter Python 3.6 or above installed on the 
+development machine. In addition, in order to build python packages you need to 
+have >= [poetry](https://python-poetry.org/) 1.1.6 package manager. Then you can 
+install and build as follows:
+```bash
+poetry install
+poetry build
+```
+
+### 2. Deploy Lua Scripts
+While deploying the project using python cli command, you can use `--develop` 
+option to state how to install all necessary Lua scripts. 
+
+```buildoutcfg
+python -m exasol_sagemaker_extension.deployment.deploy_cli \
+    --host <DB_HOST> \ 
+    --port <DB_PORT> \
+    --user <DB_USER> \
+    --pass <PASS> \
+    --schema <SCHEMA>
+    --develop
+```
+#### 2.1. Use pre-prepared Lua scripts 
+This deployment uses all the necessary scripts of the project which are already 
+prepared in the project package. The `--develop` option is not used in this 
+deployment.
+
+#### 2.2. Generate Lua scripts from scratch
+In this setup, Lua scripts are built and packaged from scratch. Therefore, this 
+deployment needs Lua interpreter Lua 5.1 or above and some Lua dependencies 
+installed on the development  machine.
+
+
+In order to be able to install the Lua dependencies, [LuaRocks](https://luarocks.org/) 
+package manager is required. You can install luarocks as follows:
+```bash
+sudo apt-get install luarocks
+```
+
+You can install the Lua dependencies in the home directory of the project
+using the following command:
+```bash
+luarocks install --only-deps *.rockspec
+```
+
+Note that, these scripts are regenerated from scratch at each commit, 
+by the help of the `regenerate_scripts.py` script executed in the `pre-commit` 
+hook script.
+
+## Running the Tests
+
+### 1. Lua Tests
+In order to run all Lua tests you can execute the following command in the project:
+```bash
+poetry run poe lua-tests
+```
+This command runs the script `./scripts/lua_tests.sh` and not only tests all Lua 
+scripts but also perform static code analysis.
+
+### 2. Python Tests
+Python tests consist of unit tests and integration tests. In the integration 
+tests [LocalStack](https://localstack.cloud/) AWS framework is used to provide 
+AWS environment. The AWS S3 integration is tested by placing Exasol database and 
+Localstack emulator on the same network. The integration test environment 
+is created with the following script:
+```bash
+./scripts/setup_integration_test.sh 
+```
+
+To run all the tests in the project use:
+```bash
+poetry run pytest tests
+```
+
+### 3. Continuous Integration (CI) Tets
```

Comment:
We probably should call them AWS Tests, because you can also run them during development, as long you have access to AWS